src/frequenz/client/reporting/_client.py

```diff
@@ -220,6 +223,8 @@ def __init__(
             ReportingStub,
             connect=connect,
             channel_defaults=channel_defaults,
+            auth_key=key,
```

Comment:
You still need to remove the ad-hoc code in this class to deal with the auth key (`self._metadata`).